### PR TITLE
Slightly simplify script/client

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: script/client [hunk 1]
```diff
@@ -230,11 +223,7 @@ sub handle_result {
 # prepend the API-base if the specified path is relative
 sub prepend_api_base {
     my $path = shift;
-
-    if ($path !~ m/^\//) {
-        $path = join('/', $options{apibase}, $path);
-    }
-    return $path;
+    return join('/', $options{apibase}, $path) if $path !~ m/^\//;
```
Comment (Martchus) [inline script/client:226]: Now simply returning `$path` when the condition is not true is obfuscated. Maybe better use the ternary operator here?

Comment (perlpunk) [inline script/client:226]: This will return the empty string if `$path =~ m/^\//`.

Comment (Martchus) [reply script/client:226]: See @perlpunk's comment. It is actually broken (and not only obfuscated).

Comment (okurz) [reply script/client:226]: Right. I could use the ternary operator

Comment (okurz) [reply script/client]: done

**Comment by okurz** [comment]
@kraih can you take a look please?

**Comment by kraih** [comment]
What's the motivation? Do you expect the client to be maintained for a long time? Doesn't this increase the maintenance burden by possibly introducing new bugs?

**Comment by okurz** [comment]
I am convinced we have a single client that we need to maintain but also rework as you proposed. Whenever we find bugs we need to add tests covering these untested areas for the future and prevent future bugs. After I was reviewing your PR about new client functionality I looked into the existing client script which is rather small and try to improve it a little but further.

File: .circleci/config.yml [hunk 2]
```diff
@@ -105,7 +105,7 @@ images:
       JUNIT_NAME_MANGLE: perl
       PERL_TEST_HARNESS_DUMP_TAP: test-results
       HARNESS: --harness TAP::Harness::JUnit --timer
-      COVEROPT: -MDevel::Cover=-select_re,'^/lib',-ignore_re,'^t/.*',+ignore_re,lib/perlcritic/Perl/Critic/Policy,-coverage,statement
+      COVEROPT: -MDevel::Cover=-select_re,'^/(lib|script)',-ignore_re,'^t/.*',+ignore_re,lib/perlcritic/Perl/Critic/Policy,-coverage,statement
```
Comment (perlpunk) [inline .circleci/config.yml:108]: Adding `script` here doesn't seem necessary, see https://github.com/os-autoinst/openQA/pull/2967

File: Makefile [hunk 3]
```diff
@@ -173,7 +173,7 @@ run-tests-within-container:
 	tools/run-tests-within-container
 
 # ignore tests and test related addons in coverage analysis
-COVER_OPTS ?= -select_re '^/lib' -ignore_re '^t/.*' +ignore_re lib/perlcritic/Perl/Critic/Policy -coverage statement
+COVER_OPTS ?= -select_re '^/(lib|script)' -ignore_re '^t/.*' +ignore_re lib/perlcritic/Perl/Critic/Policy -coverage statement
```
Comment (perlpunk) [inline Makefile:176]: same here

File: Makefile [hunk 4]
```diff
@@ -187,7 +187,7 @@ print-cover-opt:
 coverage:
 	cover ${COVER_OPTS} -test
 
-COVER_REPORT_OPTS ?= -select_re ^lib/
+COVER_REPORT_OPTS ?= -select_re ^(lib|script)/
```
Comment (perlpunk) [inline Makefile:190]: same here

Comment (okurz) [reply Makefile]: yes, I excluded that parts again for now and will leave it to https://github.com/os-autoinst/openQA/pull/2967

**Comment by okurz** [comment]
@kraih not sure if you also want to take a look at it

**Comment by kraih** [comment]
I strongly disagree with moving that horrible code into `OpenQA::Client`. It should be killed with :fire: instead.

**Comment by kraih** [comment]
And if i were to actually do a code review of this PR and point out all the flaws one by one we'd be sitting here for a week and end up with #2859. I'm not sure why you're doing this.

**Comment by okurz** [comment]
If you prefer I can exclude the "move of garbage code" and just add the test for the script in place. Would this be better? This is what I thought we had discussed earlier today.

**Comment by okurz** [comment]
The failure in OBS is due to an unrelated problem in Mojolicous-8.37 , no test failed due to changes done here. We have done a similar approach for the clone job script moving parts of the code to a lib already in 2018 and I guess we can also call the clone job related code "legacy".

**Comment by Martchus** [comment]
Maybe you can create a distinct Perl module called `OldClient.pm` (or similar) instead of mixing it into the "high quality code"™ of `Client.pm`. This way you should still be able to test it.

**Comment by okurz** [comment]
The new client is not called "New" and I think "old" and "new" can become ambiguous very soon. Same applies for "NG" ;) We have git commits and OBS versions to track our versions, not names in libraries.

**Comment by okurz** [comment]
oops, that was actually a mistake :D Sorry about that but nevertheless, we can always rename the modules. I do not see a problem with that. And as stated in before I am happy to remove the old module anytime when we have a better alternative, but not rename something "old" and "new"

**Comment by kraih** [comment]
If it was a mistake then please revert it. That code does not belong in `OpenQA::Client`. Even among our existing modules there are more fitting ones, like `OpenQA::Script`.

**Comment by okurz** [comment]
Well I had planned to merge it anyway but wanted to wait for more feedback. This doesn't break any functionality nor tests.

So I should move it to Script then I assume

**Comment by okurz** [comment]
Done in https://github.com/os-autoinst/openQA/pull/2985

**Pull request merged**